arbitrum-docs/arbos/l1-to-l2-messaging.mdx

```diff
@@ -1,21 +1,125 @@
+    <Connection from="2" to="3" label="L2-to-L1 messaging"/>
+    <Connection from="3" to="2" label="L1-to-L2 messaging"/>
+  </Nodes>
+  <NodeDescriptions>
+    <NodeDescription for='1'>Decentralized apps usually need to: <br/> - exchange information with both both layers<br/> - trust that this information exchange is trustworthy<br/> - be able to verify that their L2 transactions are settling to Ethereum L1<br/> - expose functionality to users on both layers<br/> - enjoy guarantees of bidirectional state synchronization between layers</NodeDescription>
+    <NodeDescription for='2'>L2 contracts can send messages to L1 contracts, which can trigger the execution of L1-hosted smart contracts.</NodeDescription>
+    <NodeDescription for='3'>L1 contracts can send messages to L2 contracts, which can trigger the execution of L2-hosted smart contracts.</NodeDescription>
+  </NodeDescriptions>
+</MermaidWithHtml>
+
+Without bidirectional cross-layer messaging, it would be impossible to build dApps that interact with both layers, which would severely limit the value and reach of your dApps.
+
+:::caution Editor note
+
+ - Low-confidence intro - this could be improved through more concrete / compelling examples. A "For example..." could help identify relatable situations in which these capabilities become critical. This could help make the information "sticky" for readers who are trying to solve specific problems in addition to learning how the stuff works.
+
+:::
+
+
+## 2: The conceptual primitives
+
+The rest of this concept document focuses on constructing an understanding of L1-to-L2 messaging through the following **conceptual primitives**:
+
+ 1. **Messages**: Pieces of information that are sent from one contract to another, and/or one layer to another.
+ 2. **Retryable tickets**: A type of message sent from L1 chains to L2 chains in order to trigger the execution of an L2-hosted smart contract. Think of these as special-purpose DTOs (data transfer objects) that encapsulate the information needed to execute a smart contract: the contract's address, the function to call, the function's arguments, etc.
+ 3. **Receipts**: todo
+ 4. **ETH deposits**: todo
+ 5. **Delayed Inbox**: todo
+
+
+## 3: Elaborating on the conceptual primitives
+
+Let's explore how each of these primitives relate to one another in the context of cross-layer communication.
+
+
+### 3.1: Messages
+
+ - A **message** is a data structure that contains information about a transaction that was initiated on L1, but that needs to be executed on L2. Messages are sent from L1 to L2, and they're used to trigger contract calls on L2.
+ - This primitive is important to understand if you're building a dApp that... <!-- todo -->
+
+
+:::caution Editor note
+
+ - Unsure about whether or not these things apply only to L1-to-L2 messages, or if they apply to L2-to-L1 messages as well
+ - We could evaluate consolidating cross-layer messaging into this single concept doc
```

Comment:
There's general "cross layer messaging" things we could talk about, but there's also important differences between the details of L1-to-L2 (retryables) vs. L2-to-L1 messages w/r/t Arbitrum (e.g., only retryables use address aliasing, only retryables have auto-execution, L2-to-L1 finalize differently). I think separating things into those three super-sections (x-chain messaging, L1-to-L2, L2-to-L1) would make sense (the nitro whitepaper kind of does this). A single page would probably be lengthy, but maybe that's fine?